File: 2022/15/java/Sensor.java
public class Sensor {
	private final int x;
	private final int y;
	private final int range;

	public Sensor(int x, int y, int bx, int by) {
		this.x = x;
		this.y = y;
		this.range = Math.abs(x - bx) + Math.abs(y -by);
	}

	public boolean isInRange(int x, int y) {
		return Math.abs(this.x - x) + Math.abs(this.y - y) <= this.range;
	}

	public int getX() {
		return this.x;
	}

	public int getY() {
		return this.y;
	}

	public int getRange() {
		return this.range;
	}

	@Override
	public String toString() {
		return "[Sensor] (" + x + ", " + y + ") range: " + range;
	}
}

File: 2022/15/rust/src/sensor.rs
use std::fmt::{ Display, Formatter };

pub struct Sensor {
    pub x: isize,
    pub y: isize,
    pub range: u64
}

impl Sensor {
    pub fn new(sx: isize, sy: isize, bx: isize, by: isize) -> Self {
        let range: u64 = (sx - bx).abs() as u64 + (sy - by).abs() as u64;
        Self { x: sx, y: sy, range }
    }

    pub fn is_in_range(&self, x: isize, y: isize) -> bool {
        (self.x - x).abs() as u64 + (self.y - y).abs() as u64 <= self.range
    }
}

impl Display for Sensor {
    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
        write!(f, "[Sensor] {}, {}, range: {}", self.x, self.y, self.range)
    }
}
